client/js/libs/handlebars/parse.js

```diff
@@ -1,126 +1,318 @@
 "use strict";
 
-Handlebars.registerHelper(
-	"parse", function(text) {
-		text = Handlebars.Utils.escapeExpression(text);
-		text = colors(text);
-		text = channels(text);
-		text = uri(text);
-		return text;
-	}
-);
-
-function uri(text) {
-	return window.URI.withinString(text, function(url) {
```

Comment:
I see you are getting rid of `URI.withinString` which is, I think, our only use of `URI.js` ([in our repo](https://github.com/thelounge/lounge/blob/master/client/js/libs/uri.js), [official repo](https://github.com/medialize/URI.js)). Any specific reason for this or just killing a dependency?

This seems to be a very active project, and they've just released a version that fixes [one of the bugs I noticed a long time ago](https://github.com/medialize/URI.js/issues/247#issuecomment-264720820).
The latest release packs quite a lot of changes since the version we use, see https://github.com/medialize/URI.js/compare/v1.14.1...v1.18.4.

Do you want to try running your tests against our version of `URI.js` and the latest one released?

I know we like to kill dependencies when they are tiny, but I think at that point I'd rather not roll our own URL parser.